dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/JobTask.kt

```diff
@@ -55,13 +67,55 @@ class JobTask(private val wrappedJob: JobFacade) : Ds3JobTask() {
         wrappedJob.finishedCompletable()
                 .subscribe(Action {
                 }, Consumer { throwable = it })
-        if(throwable != null) {
+        if (throwable != null) {
             throw throwable!!
         }
 
     }
 
-    override fun getJobId(): UUID = wrappedJob.jobUUID()
+    override fun getJobId(): UUID? = wrappedJob.jobUUID()
 
     public val isVisible: BooleanProperty = SimpleBooleanProperty(true)
+
+    public fun onCancelled(client: Ds3Client,
+                           type: String,
+                           log: Logger,
+                           loggingService: LoggingService,
+                           jobInterruptionStore: JobInterruptionStore,
+                           deepStorageBrowserPresenter: DeepStorageBrowserPresenter): (WorkerStateEvent) -> Unit = {
+        jobId.exists {
+            try {
+                Platform.runLater {
+                    client.cancelJobSpectraS3(CancelJobSpectraS3Request(it))
+                }
+            } catch (e: IOException) {
+                log.error("Failed to cancel $type job", e)
+                loggingService.logMessage("Could not cancel $type job", LogType.ERROR)
+            }
+            log.info("{} Job cancelled", type)
+            loggingService.logMessage("$type Job Cancelled", LogType.INFO)
```

Comment:
if the job failed to cancel and you got into the catch you are still going to get here and print that the job was canceled, maybe return in the catch or put this logic in the try block?